src/components/Canary/index.js

```diff
@@ -101,24 +105,33 @@ export function Canary({
         }
       })
       .finally(() => {
+        if (componentUnMounted) {
+          return;
+        }
         setIsLoading(false);
       });
   }, 1000);
 
-  useMemo(() => {
+  useEffect(() => {
     handleFetch();
-    const interval = setInterval(handleFetch, refreshInterval);
-    return () => clearInterval(interval);
+    const intervalRef = setInterval(handleFetch, refreshInterval);
```

Comment:
Would it be better to use `setTimeout` instead of `setInterval`? Timeout would have implicit back-pressure. With interval, there would be request every 15 sec. With timeout, next request will only happen 15s after the last one. And throttle might not be needed. 

Not a blocker. If no issues with current code, we can merge it.